sdk/src/main/java/software/amazon/awssdk/iot/discovery/DiscoveryClientConfig.java

```diff
@@ -27,6 +28,24 @@ public DiscoveryClientConfig(
         this.region = region;
         this.maxConnections = maxConnections;
         this.proxyOptions = proxyOptions;
+        this.ggServerName = "";
+    }
+
+    public DiscoveryClientConfig(
```

Comment:
Add javadoc comments to the 2 different constructors, be sure the comments on the old constructor say to use the new one if your endpoint doesn't match the certain pattern. You might want to just flat at say that the new constructor is the preferred one

Reply:
Isn't the old constructor the preferred one since most of the time they don't need a special endpoint? And it will construct it correctly for them?

Reply:
The new constructor always works 100% of the time, which is why I'd suggest it. If we'd only had people passing the whole endpoint, we wouldn't have had to do anything to support China

But it's not super important, since they're both they're now